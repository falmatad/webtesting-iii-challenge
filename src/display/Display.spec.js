// Test away!
import React from 'react';
import * as rtl from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import Display from "./Display";

test('Defaults to the word unlocked and open', async () => {

    //Set up Props
    const closed = false;
    const locked = false;

    //Bring in Component
    const displayComponent = rtl.render(<Display closed={closed} locked={locked}/>);

    //Grab Node Element from Component
    const unlocked = await displayComponent.findByText(/Unlocked/i)

    //Here is what findByText will bring back as a promise 
    console.log(unlocked.textContent);

    //Assert 
    expect(unlocked.textContent).toBe("Unlocked");

})

test("That it cannot be closed or opened if it is locked", async () => {
  
    const displayComponent = rtl.render(<Display locked={true} closed={true}/>);
    expect(displayComponent.asFragment()).toMatchSnapshot();

    const buttonDisabled = document.querySelectorAll('.led');

    console.log(buttonDisabled[0]);

    expect(buttonDisabled[0]).toHaveClass('red-led');
})