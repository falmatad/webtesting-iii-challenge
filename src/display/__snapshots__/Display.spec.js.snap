// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`That it cannot be closed or opened if it is locked 1`] = `
<DocumentFragment>
  <div
    class="display panel"
  >
    <div
      class="led red-led"
    >
      Locked
    </div>
    <div
      class="led red-led"
    >
      Closed
    </div>
  </div>
</DocumentFragment>
`;
